x-pack/legacy/plugins/epm/server/lib/kibana/index_pattern/install.ts

```diff
@@ -5,16 +5,154 @@
  */
 
 import { SavedObjectsClientContract } from 'kibana/server';
+import { KibanaAssetType } from '../../../../common/types';
+import { RegistryPackage, Dataset } from '../../../../common/types';
+import { loadFieldsFromYaml } from '../../elasticsearch/template/install';
+import * as Registry from '../../../registry';
 
-export async function installIndexPattern(
-  name: string,
+export interface Field {
+  name: string;
+  type: string;
+  required?: boolean;
+  description?: string;
+  fields?: Field[];
+}
+
+interface IndexPatternField extends Field {
+  searchable: boolean;
+  aggregatable: boolean;
+  readFromDocValues: boolean;
+}
+
+export async function installIndexPatterns(
+  pkgkey: string,
   savedObjectsClient: SavedObjectsClientContract
 ) {
-  await savedObjectsClient.create('index-pattern', getData(name));
+  const registryPackageInfo = await Registry.fetchInfo(pkgkey);
+  if (!registryPackageInfo.datasets) return;
+  const datasets = registryPackageInfo.datasets;
+  // separate logs and metrics datasets
+  const logsDatasets = datasets.filter(dataset => dataset.type === 'logs');
+  const metricsDatasets = datasets.filter(dataset => dataset.type === 'metric');
+  await createIndexPattern({
+    datasetType: 'logs',
+    datasets: logsDatasets,
+    registryPackageInfo,
+    savedObjectsClient,
+  });
+  await createIndexPattern({
+    datasetType: 'metrics',
+    datasets: metricsDatasets,
+    registryPackageInfo,
+    savedObjectsClient,
+  });
 }
 
-function getData(name: string) {
+// loop through each dataset, get all the fields, create index pattern by type.
+const createIndexPattern = async ({
+  datasetType,
+  datasets,
+  registryPackageInfo,
+  savedObjectsClient,
+}: {
+  datasetType: string;
+  datasets: Dataset[];
+  registryPackageInfo: RegistryPackage;
+  savedObjectsClient: SavedObjectsClientContract;
+}) => {
+  let allFields: Field[] = [];
```

Comment:
Not critical for this PR, but one optimization is to load the files in parallel instead of serially with something like

```typescript
const loadingFields = datasets.map(dataset => loadFieldsFromYaml(registryPackageInfo, dataset.name));
const nestedResults = await Promise.all(loadingFields);
const allFields = nestedResults.flat();
```

That way we only wait as long as the longest load time; not the sum of all the load times.